lti_consumer/lti_consumer.py

```diff
@@ -0,0 +1,829 @@
+        display_name="Request user's email",
+        # Translators: This is used to request the user's email for a third party service.
+        help="Select True to request the user's email address.",
+        default=False,
+        scope=Scope.settings
+    )
+
+    # StudioEditableXBlockMixin configuration of fields editable in Studio
+    editable_fields = (
+        'display_name', 'description', 'lti_id', 'launch_url', 'custom_parameters', 'launch_target', 'button_text',
+        'modal_height', 'modal_width', 'has_score', 'weight', 'hide_launch', 'accept_grades_past_due',
+        'ask_to_send_username', 'ask_to_send_email'
+    )
+
+    @property
+    def descriptor(self):
+        """
+        Returns this XBlock object.
+
+        This is for backwards compatibility with the XModule API.
+        Some LMS code still assumes a descriptor attribute on the XBlock object.
+        See courseware.module_render.rebind_noauth_module_to_user.
+        """
+        return self
+
+    @property
+    def context_id(self):
+        """
+        Return context_id.
+
+        context_id is an opaque identifier that uniquely identifies the context (e.g., a course)
+        that contains the link being launched.
+        """
+        return unicode(self.course_id)  # pylint: disable=no-member
+
+    @property
+    def role(self):
+        """
+        Get system user role and convert it to LTI role.
+        """
+        return ROLE_MAP.get(self.runtime.get_user_role(), u'Student')
+
+    @property
+    def course(self):
+        """
+        Return course by course id.
+        """
+        return self.runtime.descriptor_runtime.modulestore.get_course(self.course_id)  # pylint: disable=no-member
+
+    @property
+    def lti_provider_key_secret(self):
+        """
+        Obtains client_key and client_secret credentials from current course.
+        """
+        for lti_passport in self.course.lti_passports:
+            try:
+                lti_id, key, secret = [i.strip() for i in lti_passport.split(':')]
+            except ValueError:
+                _ = self.runtime.service(self, "i18n").ugettext
+                msg = _('Could not parse LTI passport: {lti_passport}. Should be "id:key:secret" string.').format(
+                    lti_passport='{0!r}'.format(lti_passport)
+                )
+                raise LtiError(msg)
+
+            if lti_id == self.lti_id.strip():
+                return key, secret
+
+        return '', ''
+
+    @property
+    def user_id(self):
+        """
+        Returns the opaque anonymous_student_id for the current user.
+        """
+        user_id = self.runtime.anonymous_student_id
+        if user_id is None:
+            raise LtiError("Could not get user id for current request")
+        return unicode(urllib.quote(user_id))
+
+    @property
+    def resource_link_id(self):
+        """
+        This is an opaque unique identifier that the TC guarantees will be unique
```

Comment:
LTI Tool Consumer, in this case, edx-platform running this XBlock. Updating the docs.